src/mito2/src/engine.rs

```diff
@@ -128,6 +130,9 @@ impl MitoEngine {
         object_store_manager: ObjectStoreManagerRef,
         schema_metadata_manager: SchemaMetadataManagerRef,
         plugins: Plugins,
+        #[cfg(feature = "enterprise")] extension_range_provider_factory: Option<
+            BoxedExtensionRangeProviderFactory,
+        >,
```

Comment:
Can we avoid adding this to the parameter? We will have the same issue in https://github.com/GreptimeTeam/greptimedb/pull/6391

Can we get this from the plugins?